### PR TITLE
Multishipping Checkout Store id is saved NULL in sales_order_item Table: FIXED

### PR DESCRIPTION
**Pull request by mahesh-rajawat**
During Multishipping Checkout Store Id is saving NULL in **sales_order_item** Table. Fixed in this PR.

### Description (*)

I was fixing the Quote item id issue in this PR #19192 , found there is also Store id issue with Multishipping checkout. There was no **store_id** column in **quote_address_item**  that's why
store_id values saved NULL. I have added **store_id** column in **quote_address_item** table in this PR.
### Fixed Issues (if relevant)

1. 
2. 

### Manual testing scenarios (*)

1. After updating code i can check store_id saved correctly with **Multishipping Checkout**  https://prnt.sc/llfjzv 
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @maheshWebkul721. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-3515](https://jira.corp.magento.com/browse/ENGCOM-3515) has been created to process this Pull Request

File: app/code/Magento/Quote/etc/db_schema.xml [hunk 1]
```diff
@@ -352,6 +352,8 @@
                 comment="Super Product Id"/>
         <column xsi:type="int" name="parent_product_id" padding="10" unsigned="true" nullable="true" identity="false"
                 comment="Parent Product Id"/>
+        <column xsi:type="smallint" name="store_id" padding="5" unsigned="true" nullable="true" identity="false"
```
Comment (sidolov) [inline app/code/Magento/Quote/etc/db_schema.xml:355]: Unfortunately, we can't add a foreign key to store table due to [Split database performance solution](https://devdocs.magento.com/guides/v2.3/config-guide/multi-master/multi-master.html). This solution provides an ability to move quote related tables to a separate database and with current changes, we will have the fatal error due to missing table.

**Review by sidolov — CHANGES_REQUESTED** [review]

File: app/code/Magento/Quote/etc/db_schema.xml (hunk 1 above)
Comment (mahesh-rajawat) [reply app/code/Magento/Quote/etc/db_schema.xml:355]: @sidolov , foreign key removed

**Comment by mahesh-rajawat** [comment]
@sidolov  any update on it?

**Comment by sivaschenko** [comment]
Thanks for the update @maheshWebkul721 ! We are continuing to process this pull request further.

**Comment by magento-engcom-team** [comment]
Hi @maheshWebkul721. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Pull request merged**